android/src/main/java/io/ably/flutter/plugin/AblyMethodCallHandler.java

```diff
@@ -43,29 +43,28 @@
 public class AblyMethodCallHandler implements MethodChannel.MethodCallHandler {
   private Context applicationContext;
 
-  public interface HotRestartCallback {
-    // Called on every fresh start and on every hot restart
+  public interface OnClearPlatformInstances {
     void on();
```

Comment:
I think this method name can be renamed to suit better.

For example: `HotRestartListener.onHotRestart` would have made more sense with older class name

Reply:
If we observe the behaviour of `registerAbly`, this dart side makes the method call to clear instances of clients in the platform side. This method calls the `HotRestartCallback`. I think it's not immediately obvious that when we register the plugin, we run the hot restart callback, and why.

It actually doesn't happen only onHotRestart, it happens whenever the platform method is called. Since the platform method clears instances of clients, I renamed `registerAbly` to `clearPlatformInstances`. This callback interface does exactly that, it clears platform instances.

Reply:
I have renamed it to `ResetAblyClientsCallback`. I think this makes sense. What do you think?

Reply:
@ben-xD Reading @tiholic's opening comment for this conversation, he's commenting on the naming of the _method_, not the _interface_. I agree, that `on` is a very opaque _method_ name - albeit inherited by you from the previous state of the codebase - but perhaps changing meaning now that you've renamed the owning interface. So, I think there's still work to do here to address his comment, please.

Reply:
I've renamed `on` to `run`, so it reads like this: `resetAblyClientsCallback.run();`. I'd like to improve it but that's the best I have so far. bf01732c9b90ecdc5168b78eceea20aee807cb33

Reply:
That is clearer, thanks.